lib/ecto/query/builder/order_by.ex

```diff
@@ -128,6 +131,16 @@ defmodule Ecto.Query.Builder.OrderBy do
 
   defp to_field(field), do: {{:., [], [{:&, [], [0]}, field]}, [], []}
 
+  @doc """
+  Called at runtime to verify the ordering mode
+  """
+  def mode!(nil), do: @default_mode
```

Comment:
Have this function receive opts as argument to we extract at runtime rather than compile time.